src/server/game/Maps/Map.cpp

```diff
@@ -736,6 +740,23 @@ void Map::VisitNearbyCellsOf(WorldObject* obj, TypeContainerVisitor<Acore::Objec
     }
 }
 
+void Map::UpdatePlayerZoneStats(uint32 oldZone, uint32 newZone)
+{
+    // Nothing to do if no change
+    if (oldZone == newZone)
+        return;
+
+    if (oldZone != MAP_INVALID_ZONE)
+    {
+        uint32& oldZoneCount = _zonePlayerCountMap[oldZone];
```

Comment:
I'm confused about what happens when the first player gets added to the map. Will this log an error immediately as this returns 0 if the entry does not exist?

Reply:
this doesnt fire when adding, its the remove block (updating the old zone), which should never be empty
trinitycore forces an assert here (crash) if it is empty, so should never happen
however AC doesnt like asserts